cfme/tests/cloud/test_keypairs.py

```diff
@@ -73,41 +78,54 @@ def test_keypair_crud_with_key(provider, appliance):
 @pytest.mark.tier(3)
 def test_keypair_create_cancel(provider, appliance):
     """ This will test cancelling on adding a keypair
-
-    Steps:
-        * Provide Keypair name.
-        * Select Cloud Provider.
-        * Also delete it.
-
     Polarion:
-        assignee: mnadeem
+        assignee: mmojzis
         casecomponent: WebUI
+        caseimportance: low
         initialEstimate: 1/4h
+        testSteps:
+            1. Cancel creating keypair.
     """
     keypair = appliance.collections.cloud_keypairs.create(
         name="",
         provider=provider,
         cancel=True
     )
-
     assert not keypair.exists
 
 
+@pytest.mark.provider([EC2Provider], override=True, scope="module", selector=ONE_PER_TYPE)
+@pytest.mark.tier(3)
+def test_keypair_create_validation(provider, appliance):
+    """ This will test validating the new of name of the key pair
+    Polarion:
+        assignee: mmojzis
+        casecomponent: WebUI
+        caseimportance: low
+        initialEstimate: 1/4h
+        testSteps:
+            1. Try to add key pair with empty name.
+    """
+    keypair_collection = appliance.collections.cloud_keypairs
+    view = navigate_to(keypair_collection, 'Add')
+    view.fill({'provider': provider})
+    assert not view.form.add.active
```

Comment:
**Question** I think this test could be extended a bit more, perhaps with invalid key pair names? Thoughts?

Reply:
Extended for invalid key pair name but that could be tested only on OpenStack. Also added test that checks public key.